Trading/Mode/index_trading_mode/index_trading.py

```diff
@@ -448,6 +457,35 @@ def _get_filtered_traded_coins(self):
             )))
         return []
 
+    def _get_supported_distribution(self) -> list:
+        if self.trading_config[IndexTradingModeProducer.INDEX_CONTENT]:
+            full_distribution = self.trading_config[IndexTradingModeProducer.INDEX_CONTENT]
+            traded_bases = set(
+                symbol.base
+                for symbol in self.exchange_manager.exchange_config.traded_symbols
+            )
+            distribution = [
+                asset
+                for asset in full_distribution
+                if asset[index_distribution.DISTRIBUTION_NAME] in traded_bases
+            ]
+            if removed_assets := [
+                asset[index_distribution.DISTRIBUTION_NAME]
+                for asset in full_distribution
+                if asset not in distribution
+            ]:
+                self.logger.info(
+                    f"Ignored {len(removed_assets)} assets {removed_assets} from configured "
+                    f"distribution as absent from traded pairs."
+                )
+            return distribution
```

Comment:
Should we add a min assets count check like raising an error if the index that should contains 5 assets only has 2 remaining coins after the asset removal process ?

Reply:
currently it errors if there is less than 1 element, we could make this min configurable yep, and we set the last activity to "not enough tradable assets" or something like this

Reply:
Yes I think we should raise something only to alert us if it happens. I'm not sure we should block the bot if the assets count is >= 1

Reply:
It could be an automatic value of now like if more than 50% of the assets are disabled

Reply:
right, good idea, i'll add an error log

Reply:
up !